airbyte-integrations/connectors/source-instagram/source_instagram/client/api.py

```diff
@@ -363,7 +356,18 @@ def _get_insights(self, item) -> Iterator[Any]:
         else:
             metrics = self.MEDIA_METRICS
 
-        return item.get_insights(params={"metric": metrics})
+        try:
+            return item.get_insights(params={"metric": metrics}), True
+        except FacebookRequestError as error:
+            # An error might occur if the media was posted before the most recent time that
+            # the user's account was converted to a business account from a personal account
+            if error.api_error_subcode() == 2108006:
+                logger.error(f"Insights error for business_account_id {account_id}: {error.body()}")
+
+                # We receive all Media starting from the last one, and if on the next Media we get an Insight error,
+                # then no reason to make inquiries for each Media further, since they were published even earlier.
+                return None, False
```

Comment:
why we return tuple?

Reply:
agree, made changes.
But I set the check to the `is None` value. Becausethe media may not have insights and an empty array will be returned, this does not mean that we need to break the loop for the all media of this account